### PR TITLE
API 6.9

### PR DESCRIPTION
**Pull request by harshil21**
### Checklist

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- [x] Added myself alphabetically to ``AUTHORS.rst`` (optional)
- [x] Added new classes & modules to the docs and all suitable ``__all__`` s
- [x] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior
- [x] Create PR to remove functionality deprecated in this PR

**If the PR contains API changes (otherwise, you can ignore this passage)**

- [x] Checked the Bot API specific sections of the [Stability Policy](https://docs.python-telegram-bot.org/stability_policy.html)

- New classes:

   - [x] Added ``self._id_attrs`` and corresponding documentation
   - [x] ``__init__`` accepts ``api_kwargs`` as kw-only

- Added new shortcuts:

   - [x]  In :class:`~telegram.Chat` & :class:`~telegram.User` for all methods that accept ``chat/user_id``
   - [x]  In :class:`~telegram.Message` for all methods that accept ``chat_id`` and ``message_id``
   - [x]  For new :class:`~telegram.Message` shortcuts: Added ``quote`` argument if methods accepts ``reply_to_message_id``
   - [x] In :class:`~telegram.CallbackQuery` for all methods that accept either ``chat_id`` and ``message_id`` or ``inline_message_id``

-  If relevant:

   - [x] Added new constants at :mod:`telegram.constants` and shortcuts to them as class variables
   - [x] Link new and existing constants in docstrings instead of hard-coded numbers and strings
   - [x]  Add new message types to :attr:`telegram.Message.effective_attachment`
   - [x] Added new handlers for new update types
   - [x] Add the handlers to the warning loop in the :class:`~telegram.ext.ConversationHandler`
   - [x] Added new filters for new message (sub)types
   - [x]  Added or updated documentation for the changed class(es) and/or method(s)
   - [x] Added the new method(s) to ``_extbot.py``
   - [x] Added or updated ``bot_methods.rst``
   - [x] Updated the Bot API version number in all places: ``README.rst`` and ``README_RAW.rst`` (including the badge), as well as ``telegram.constants.BOT_API_VERSION_INFO``
   - [x] Added logic for arbitrary callback data in :class:`telegram.ext.ExtBot` for new methods that either accept a ``reply_markup`` in some form or have a return type that is/contains :class:`~telegram.Message`

### CODE REVIEW

**Comment by Poolitzer** [comment]
Not sure with the versionchanged text at the promote shortcut and not sure if there should be a versionchanged within the message class where only the docstring was extended.

File: telegram/_chat.py [hunk 1]
```diff
@@ -938,9 +941,12 @@ async def promote_member(
 
         .. versionadded:: 13.2
         .. versionchanged:: 20.0
-           The argument ``can_manage_voice_chats`` was renamed to
+           The argument `can_manage_voice_chats`` was renamed to
            :paramref:`~telegram.Bot.promote_chat_member.can_manage_video_chats` in accordance to
            Bot API 6.0.
+        .. versionchanged:: NEXT.VERSION
+           The arguments `can_post_stories`, `can_edit_stories` and `can_delete_stories` were
+           added.
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply telegram/_chat.py:949]: Not necessarily, but it also doesn't hurt to have IMO :)

File: telegram/_chatadministratorrights.py [hunk 2]
```diff
@@ -148,6 +178,9 @@ def __init__(
         self.can_post_messages: Optional[bool] = can_post_messages
         self.can_edit_messages: Optional[bool] = can_edit_messages
         self.can_pin_messages: Optional[bool] = can_pin_messages
+        self.can_post_stories: Optional[bool] = can_post_stories
+        self.can_edit_stories: Optional[bool] = can_edit_stories
+        self.can_delete_stories: Optional[bool] = can_delete_stories
         self.can_manage_topics: Optional[bool] = can_manage_topics
 
         self._id_attrs = (
```
Comment (Bibo-Joshi) [inline telegram/_chatadministratorrights.py:186]: I guess the id attrs (and the corresponding docstring) should be updated as well. This is in line with our stability policy 😎

File: telegram/_writeaccessallowed.py [hunk 3]
```diff
@@ -26,28 +26,54 @@
 class WriteAccessAllowed(TelegramObject):
```
Comment (Bibo-Joshi) [inline telegram/_writeaccessallowed.py:26]: Do we want to add id_attrs ot this class? I guess `web_app_name` is a must have. `from_request` and `from_attachment_menu` can make sense, but I wouldn't insist on them. If we don't use them for equality comparison, then this should go into a standalone PR …

File: telegram/_writeaccessallowed.py [hunk 4]
```diff
@@ -26,28 +26,54 @@
...
     This object represents a service message about a user allowing a bot to write messages after
-    adding the bot to the attachment menu or launching a Web App from a link.
+    adding the bot to the attachment menu, launching a Web App from a link, or accepting an
+    explicit request from a Web App sent by the method
+    `requestWriteAccess <https://core.telegram.org/bots/webapps#initializing-mini-apps>`_.
 
     .. versionadded:: 20.0
 
     Args:
         web_app_name (:obj:`str`, optional): Name of the Web App which was launched from a link.
```
Comment (Bibo-Joshi) [inline telegram/_writeaccessallowed.py:36]: the docstring changed

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the nice PR! I left a couple of minor comments. Additionally, kindly double check with the changes in https://t.me/bot_api_changes/96 & 97. I saw one change in WriteAccessAllowed & there are also a couple of changes in the `can_*` argument docstrings. Or did everything of that go into doc-fixes alngo with https://github.com/python-telegram-bot/python-telegram-bot/commit/3c2642f566fb967d667a2592a11536c7dfade033 ?

File: telegram/_chatadministratorrights.py (hunk 2 above)
Comment (harshil21) [reply telegram/_chatadministratorrights.py:186]: can you remind me why we are adding optional attributes to _id_attrs again?

Comment (Bibo-Joshi) [reply telegram/_chatadministratorrights.py:186]: Mh, I had indeed missed that only some if the attributes are optional. But then again, we already started with them, so we might as well continue...

File: telegram/_writeaccessallowed.py (hunk 3 above)
Comment (Poolitzer) [reply telegram/_writeaccessallowed.py:26]: Wdym "not use them for equality comparison", thats the point of id_attrs, no?

Comment (Bibo-Joshi) [reply telegram/_writeaccessallowed.py:26]: No, that's what I meant: I don't insist on adding `from_*` to `_id_attrs` and if we don't, then the change (adding only `web_app_name`) should go into a different pr

Comment (harshil21) [reply telegram/_writeaccessallowed.py:26]: yea I would prefer having just `web_app_name` in a new pr

Comment (Bibo-Joshi) [reply telegram/_writeaccessallowed.py:26]: #3909

**Review by Bibo-Joshi — APPROVED** [review]

**Comment by Bibo-Joshi** [comment]
@harshil21 @Poolitzer is there anything to add from your side?

**Comment by harshil21** [comment]
Nope, LGTM
